### ISSUE TITLE
[Feature] Federate metadata

### ISSUE THREAD

**Issue by kroese** [LemmyNet/lemmy#4598]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?

### Is your proposal related to a problem?

I always assumed my Lemmy instance was only making connections to other instances, but  its constantly making GET requests to all kinds of non-Lemmy webservers for scraping.

I am not sure if Im too happy about that, for example Reddit may rate-limit my IP, other websites might block me for constant scraping and it might connect to all kinds of shady or illegal websites. Also, if the number of instances grows, that would trigger a DoS attack at the external URL by all those instances fetching the metadata at roughly the same time.

Could there be an option to disable metadata scraping for federated posts? I dont mind my server connecting to other Lemmy servers, I just want it to trust the received metadata (thumbnail url and summary) instead of crawling the whole web just to verify it. For local posts the metadata fetching is fine.

### Describe the solution you'd like.

A setting to disable the fetching of metadata for federated (remote) posts, and trust those fields from the remote instance.

### Describe alternatives you've considered.

N/A

**Comment by dessalines** [comment]
I can't really figure out the root reason we do this at all. [Here's the line](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/objects/post.rs#L275), and these are the potential fields that are updated in the metadata fetch.

![Screenshot_20240405_133100_Termux](https://github.com/LemmyNet/lemmy/assets/930215/47d1c4c7-c2d5-4c4b-8cd5-cd56e697a226)

I think `Page` already has the thumbnail_url, but the others we might be able to federate as attachments or something, if they aren't already? 

cc @Nutomic

**Comment by Nutomic** [comment]
The thumbnail is stored which afaik is also used by other Activitypub platforms. Afaik other platforms including Mastodon dont federate link metadata in any way, but we could do it by adding some fields in [Link](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/protocol/objects/page.rs#L75). The only concern I can think of is that a malicious instance could put wrong info into the metadata.

**Comment by Nutomic** [comment]
> This would already increase federating performance (and privacy!) by a great amount. Because as described in other issues, the reason why people lag behind can be of the slow metadata fetching.

This is already fixed by https://github.com/LemmyNet/lemmy/pull/4564. But federating the embed info makes sense.

**Comment by Nutomic** [comment]
Relevant FEP: https://codeberg.org/fediverse/fep/src/branch/main/fep/8967/fep-8967.md

**Comment by Trying2KnowMyself** [comment]
This also impacts posts to remote comms. If I set a thumbnail URL on a post, it’s only honored by the instance I’m posting from, and not the instance I’m posting to.

**Comment by kroese** [comment]
@Trying2KnowMyself Yes, I reported that problem of the thumbnail not federating multiple times, but it seems to have no priority at all.

**Comment by Trying2KnowMyself** [comment]
I unfortunately continue to learn that

> federation sucks and I hate it.

Is truer than I knew when I first said it.

**Comment by kroese** [comment]
That's why I moved to [Piefed](https://piefed.social/). Not just because it federates custom thumbnails correctly, but because their dev would fix it quickly when it did not.

**Comment by Trying2KnowMyself** [comment]
Python is my second least favorite language, right after English, so even if rimu weren’t a lib and a Zionist I’d probably pass on that.

**Comment by codeinabox** [comment]
I was unaware of this issue until I asked why someone why their blog is now blocking Lemmy, and they offered [this explanation](https://mastodonapp.uk/@ianvisits/117071182927542901):

> Because Lemmy keeps trying to crash my website when links are shared on it, as it's such an aggressive crawler, and I had to put a Cloudflare bot block.
